### ISSUE TITLE
Flight model diagnostic

### ISSUE THREAD

**Issue by brunofalmeida** [HeronMkII/eps#35]
Update `harness_tests/diagnostic_test` for the flight model configuration. See the "BUS Architecture Specification" document ("5 Diagnostic Tests" section) for what needs to be checked, and the "Command Protocol Specification" document for the CAN field numbers.

**The purpose of these diagnostic tests is to verify that all the hardware is working, i.e. all measurements are nominal and special actions/controls have the intended effects.**

Most of the test is checking all of the housekeeping and optical data fields, where the general process is:
- Enqueue a CAN message
- Process the CAN message
- Get the response CAN message and extract the data from it
- Apply a conversion function to the raw data to get a "real" value (e.g. a voltage)
- Check the value is greater than the minimum value and less than the maximum value

Basically you should try to use CAN messages for everything (because this is how OBC will control it in the final implementation).

For each field/type of data that you need to convert to a usable value, try to figure out the appropriate conversion function in `lib-common/src/conversions.c`. If you're not sure which to use, just take your best guess and we can fix it later if necessary.

The "control" tests involve sending CAN messages of the EPS_CTRL type, so again you should use CAN to control those actions.
